Assets/MixedRealityToolkit.SDK/Features/Utilities/StabilizationPlaneModifier.cs

```diff
@@ -0,0 +1,390 @@
+                    if (targetOverride)
+                    {
+                        targetOverridePreviousPosition = targetOverride.position;
+                    }
+                }
+            }
+        }
+
+        [SerializeField, Tooltip("Keeps track of position-based velocity for the target object.")]
+        private bool trackVelocity;
+        public bool TrackVelocity
+        {
+            get
+            {
+                return trackVelocity;
+            }
+            set
+            {
+                trackVelocity = value;
+                if (TargetOverride)
+                {
+                    targetOverridePreviousPosition = TargetOverride.position;
+                }
+            }
+        }
+
+        [Tooltip("Default distance to set plane if plane is gaze-locked or if no object is hit.")]
+        public float DefaultPlaneDistance = 2.0f;
+
+        [Tooltip("Visualize the plane at runtime.")]
+        public bool DrawGizmos;
+        
+        [SerializeField, Tooltip("Override plane to use. Usually used to set plane to a slate like a menu")]
+        private StabilizationPlaneOverride overridePlane;
+
+        public StabilizationPlaneOverride OverridePlane
+        {
+            get
+            {
+                return overridePlane;
+            }
+            set
+            {
+                overridePlane = value;
+            }
+        }
+
+        private IMixedRealityInputSystem inputSystem = null;
+
+        /// <summary>
+        /// The active instance of the input system.
+        /// </summary>
+        private IMixedRealityInputSystem InputSystem
+        {
+            get
+            {
+                if (inputSystem == null)
+                {
+                    MixedRealityServiceRegistry.TryGetService<IMixedRealityInputSystem>(out inputSystem);
+                }
+                return inputSystem;
+            }
+        }
+
+        /// <summary>
+        /// Position of the plane in world space.
+        /// </summary>
+        private Vector3 planePosition;
+
+        /// <summary>
+        /// Current distance of the plane from the user's head. Only used when not using the target override
+        /// or the GazeManager to set the plane's position. 
+        /// </summary>
+        private float currentPlaneDistance = 4.0f;
+
+        /// <summary>
+        /// Tracks the previous position of the target override object. Used if velocity is being tracked.
+        /// </summary>
+        private Vector3 targetOverridePreviousPosition;
+
+        /// <summary>
+        /// Used for representing latest plane drawn as gizmo
+        /// </summary>
+        private StabilizationPlaneOverride debugPlane;
+
+        /// <summary>
+        /// Used for representing the debug mesh
+        /// </summary>
+        private GameObject debugMesh;
+
+        private void Awake()
+        {
+#if UNITY_EDITOR
+            debugMesh = GameObject.CreatePrimitive(PrimitiveType.Quad);
```

Comment:
Since this is not active whole time, add hideflag to hide in hierarchy?
https://docs.unity3d.com/ScriptReference/HideFlags.html